csm/core/services/firmware_update.py

```diff
@@ -40,17 +40,23 @@ async def upload_package(self, package_ref, filename):
         :param filename: str
         :return: dict
         """
-        firmware_package_path = package_ref.save_file(self._fw_storage_path, filename, True)
+        try:
+            firmware_package_path = package_ref.save_file(self._fw_storage_path, filename, True)
+        except Exception as e:
+            Log.error(f'Failed to save the package: {e}')
+            raise CsmInternalError(f'Failed to save the package: {e}')
         model = await self._update_repo.get_current_model(const.FIRMWARE_UPDATE_ID)
         if model and model.is_in_progress():
             raise InvalidRequest("You can't upload a new package while there is an ongoing update")
 
+        Log.debug("Saving model for firmware update")
         model = UpdateStatusEntry.generate_new(const.FIRMWARE_UPDATE_ID)
         model.version = os.path.splitext(os.path.basename(filename))[0]
         model.description = os.path.join(self._fw_storage_path, filename)
         model.file_path = os.path.join(self._fw_storage_path, filename)
         model.details = ''
         model.mark_uploaded()
+        Log.debug(model.to_printable())
```

Comment:
You can move the above DEBUG statement here

Reply:
Acknowledged.

Reply:
Comment resolved